src/systems/filecoin_markets/storage_market/storage_market_actor_code.go

```diff
@@ -225,11 +238,15 @@ func (a *StorageMarketActorCode_I) ProcessDealSlash(rt Runtime, dealIDs []deal.D
 }
 
 func (a *StorageMarketActorCode_I) ProcessDealPayment(rt Runtime, dealIDs []deal.DealID, newPaymentEpoch block.ChainEpoch) {
+	TODO() // verify caller must be StorageMinerActor
+
 	h, st := a.State(rt)
 
 	for _, dealID := range dealIDs {
-		deal := st._safeGetOnChainDeal(rt, dealID)
-		st._assertActiveDealState(rt, dealID)
+		deal, ok := st._getOnChainDeal(dealID)
+		if !ok {
+			continue
```

Comment:
Is this failure allowed (e.g. because the deal has already expired, but we are computing expiration lazily)? If so, would be good to add a comment

Reply:
yes, it depends on if we delete the deal when it expires which i think we should. This is left for a follow-up.